src/plugin/systems/mod.rs

```diff
@@ -347,19 +338,13 @@ pub mod tests {
     pub struct HeadlessRenderPlugin;
 
     impl Plugin for HeadlessRenderPlugin {
-        fn build(&self, app: &mut App) {
-            app.add_plugins((
-                AssetPlugin::default(),
-                ScenePlugin,
-                RenderPlugin {
-                    render_creation: RenderCreation::Automatic(WgpuSettings {
-                        backends: None,
-                        ..Default::default()
-                    }),
-                    ..Default::default()
-                },
-                ImagePlugin::default(),
-            ));
+        fn build(&self, _app: &mut App) {
+            #[cfg(feature = "async-collider")]
+            {
+                use bevy::{asset::AssetPlugin, render::mesh::MeshPlugin, scene::ScenePlugin};
```

Comment:
We might want to add those directly within RapierPhysicsPlugin when we use the feature `async-collider`.